sklearn/neighbors/_base.py

```diff
@@ -694,6 +695,10 @@ def _fit(self, X, y=None):
                     "try algorithm='ball_tree' "
                     "or algorithm='brute' instead."
                 )
+            self._sp_tree = spKDTree(
+                X,
+                self.leaf_size,
+            )
```

Comment:
Replacing `_tree` itself was *much* messier, so I tried to keep this scoped for prototyping.